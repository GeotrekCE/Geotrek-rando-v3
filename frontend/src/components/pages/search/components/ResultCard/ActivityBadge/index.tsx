import React from 'react';
import styled from 'styled-components';
import SVG from 'react-inlinesvg';

import { colorPalette, getSpacing } from 'stylesheet';

interface Props {
  iconUri: string;
  className?: string;
}

export const ActivityBadge: React.FC<Props> = ({ iconUri, className }) => {
  return (
    <Container className={className}>
      <SVG src={iconUri} className="w-5" />
    </Container>
  );
};

const Container = styled.div`
  width: ${getSpacing(8)};
  height: ${getSpacing(8)};
  border-radius: 50%;

  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  border: 2px solid ${colorPalette.white};
  background-color: ${colorPalette.primary1};
  color: ${colorPalette.white};
`;
