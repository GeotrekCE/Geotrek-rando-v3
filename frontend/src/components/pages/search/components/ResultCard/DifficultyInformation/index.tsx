import React from 'react';
import SVG from 'react-inlinesvg';

import { colorPalette, getSpacing, typography } from 'stylesheet';
import styled from 'styled-components';

interface Props {
  iconUri: string;
  children: React.ReactNode;
  className?: string;
}

export const DifficultyInformation: React.FC<Props> = ({ iconUri, children, className = '' }) => {
  return (
    <div className={`flex items-center text-primary1 ${className} `}>
      <SVG src={iconUri} className="desktop:w-6" />
      <InformationText>{children}</InformationText>
    </div>
  );
};

const InformationText = styled.span`
  margin-left: ${getSpacing(2)};
  ${typography.small}
  color: ${colorPalette.darkPurple}
`;
