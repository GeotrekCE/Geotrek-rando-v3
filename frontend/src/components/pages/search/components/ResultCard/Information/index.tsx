export { LocalIconInformation } from './LocalIconInformation';
export { RemoteIconInformation } from './RemoteIconInformation';
