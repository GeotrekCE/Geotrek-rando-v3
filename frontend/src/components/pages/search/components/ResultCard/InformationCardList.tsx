import { InformationCard } from 'modules/results/interface';
import { InformationCardItem } from '../InformationCardItem';

interface InformationsListProps {
  informations?: InformationCard[];
}

export const InformationCardList: React.FC<InformationsListProps> = ({ informations }) => {
  if (!informations || informations.length === 0) {
    return null;
  }
  return (
    <ul className="mt-4 flex flex-wrap gap-2 desktop:gap-4">
      {informations.map(item => (
        <li key={item.label}>
          <InformationCardItem {...item} />
        </li>
      ))}
    </ul>
  );
};
