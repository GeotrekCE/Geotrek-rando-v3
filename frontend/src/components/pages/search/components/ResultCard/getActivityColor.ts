import getConfig from 'next/config';

const default_trek_color = '#001B84';
const default_outdoor_color = '#E69736';
const default_touristic_event_color = '#62AB41';
const default_touristic_content_color = '#3B89A2';

const getActivityColor = (type?: string | null): string => {
  const {
    publicRuntimeConfig: {
      colors,
      global: { groupTreksAndOutdoorFilters },
    },
  } = getConfig();

  const defaultColor = colors.primary1?.DEFAULT || '#AA397D';

  const outdoorColor =
    groupTreksAndOutdoorFilters === true
      ? colors?.categories?.trek || default_trek_color
      : colors?.categories?.outdoor || default_outdoor_color;

  const color =
    (type
      ? {
          TREK: colors?.categories?.trek || default_trek_color,
          OUTDOOR_SITE: outdoorColor,
          OUTDOOR_COURSE: outdoorColor,
          TOURISTIC_CONTENT: colors?.categories?.service || default_touristic_content_color,
          TOURISTIC_EVENT: colors?.categories?.events || default_touristic_event_color,
          POI: colors?.categories?.outdoor || default_outdoor_color,
          practices: colors?.categories?.trek || default_trek_color,
          outdoorPractice: colors?.categories?.outdoor || default_outdoor_color,
          event: colors?.categories?.events || default_touristic_event_color,
          categories: colors?.categories?.service || default_touristic_content_color,
        }[type]
      : defaultColor) ?? defaultColor;

  return color;
};

export default getActivityColor;
