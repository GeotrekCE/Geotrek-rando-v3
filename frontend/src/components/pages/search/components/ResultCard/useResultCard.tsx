import { routes } from 'services/routes';

export const useResultCard = (id: number, title: string) => {
  const titleWithNoSpace = title.replace(/ /g, '-');
  const detailsPageUrl = `${routes.DETAILS}-${id}-${encodeURI(titleWithNoSpace)}`;
  return { detailsPageUrl };
};
