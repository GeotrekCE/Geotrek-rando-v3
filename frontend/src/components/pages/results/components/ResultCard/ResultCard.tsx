import React from 'react';
import styled, { css } from 'styled-components';

import { borderRadius, colorPalette, desktopOnly, getSpacing, typography } from 'stylesheet';
import { flexGap } from 'services/cssHelpers';

import { Chip } from 'components/Chip';
import { Button } from 'components/Button';

import { Clock } from 'components/Icons/Clock';
import { CodeBrackets } from 'components/Icons/CodeBrackets';
import { TrendingUp } from 'components/Icons/TrendingUp';
import { Walking } from 'components/Icons/Walking';

import { Information } from './Information';
import { ActivityBadge as RawActivityBadge } from './ActivityBadge';

export const ResultCard: React.FC = () => {
  return (
    <Container>
      <ImageContainer>
        <ActivityBadge icon={Walking} />
      </ImageContainer>
      <DetailsContainer>
        <DetailsLayout>
          <Place>Saint-Etienne-du-Valdonnez</Place>
          <Title>Balade au pays des menhirs</Title>
          <TagContainer>
            <TagLayout>
              <Chip>En famille</Chip>
              <Chip>Ciel étoilé</Chip>
              <Chip>En famille</Chip>
            </TagLayout>
          </TagContainer>
          <InformationContainer>
            <InformationLayout>
              <Information icon={Clock}>2h</Information>
              <Information icon={CodeBrackets}>5km</Information>
              <Information icon={TrendingUp}>+360m</Information>
            </InformationLayout>
          </InformationContainer>
        </DetailsLayout>
        <BookingButtonContainer>
          <Button>Réserver</Button>
        </BookingButtonContainer>
      </DetailsContainer>
    </Container>
  );
};

const Container = styled.div`
  display: flex;
  flex-direction: column;

  margin: ${getSpacing(4)};

  border-radius: ${borderRadius.card};
  overflow: hidden;

  ${desktopOnly(
    css`
      flex-direction: row;
      margin: ${getSpacing(6)};
    `,
  )}
`;

const ImageContainer = styled.div`
  height: ${getSpacing(31)};
  width: 100%;

  background-image: url('images/hiking-cover.jpg');
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center center;

  position: relative;

  ${desktopOnly(
    css`
      height: auto;
      max-width: ${getSpacing(56)};
      flex-shrink: 0;
    `,
  )}
`;

const DetailsContainer = styled.div`
  display: flex;

  padding: ${getSpacing(4)};

  border: 1px solid ${colorPalette.greySoft};
  border-top: none;
  border-radius: 0 0 ${borderRadius.card} ${borderRadius.card};

  ${desktopOnly(
    css`
      padding: ${getSpacing(6)};

      border: 1px solid ${colorPalette.greySoft};
      border-left: none;
      border-radius: 0 ${borderRadius.card} ${borderRadius.card} 0;
    `,
  )}
`;

const DetailsLayout = styled.div`
  display: flex;
  flex-direction: column;
`;

const BookingButtonContainer = styled.div`
  margin-left: ${getSpacing(4)};
  display: none;

  ${desktopOnly(
    css`
      display: block;
    `,
  )}
`;

const Place = styled.span`
  ${typography.small}
`;

const Title = styled.span`
  margin-top: ${getSpacing(1)};

  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;

  ${typography.h1}
  color: ${colorPalette.primary1};

  ${desktopOnly(
    css`
      text-overflow: clip;
      white-space: normal;
    `,
  )}
`;

const TagContainer = styled.div`
  display: flex;
  margin-top: ${getSpacing(2)};

  ${desktopOnly(
    css`
      margin-top: ${getSpacing(4)};
    `,
  )}
`;

const TagLayout = styled.div`
  ${flexGap(getSpacing(2))}

  ${desktopOnly(
    css`
      ${flexGap(getSpacing(4))}
    `,
  )}
`;

const InformationContainer = styled.div`
  margin-top: ${getSpacing(4)};
`;

const InformationLayout = styled.div`
  ${flexGap(getSpacing(4))}
`;

const ActivityBadge = styled(RawActivityBadge)`
  position: absolute;
  top: ${getSpacing(4)};
  left: ${getSpacing(4)};
`;
