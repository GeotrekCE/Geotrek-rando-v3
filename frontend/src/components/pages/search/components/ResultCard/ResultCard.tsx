import { Altitude } from 'components/Icons/Altitude';
import { Calendar } from 'components/Icons/Calendar';
import { Height } from 'components/Icons/Height';
import { Modal } from 'components/Modal';
import { DetailsCoverCarousel } from 'components/pages/details/components/DetailsCoverCarousel';
import React, { useContext } from 'react';
import styled, { css } from 'styled-components';
import { FormattedMessage, useIntl } from 'react-intl';

import { borderRadius, colorPalette, desktopOnly, getSpacing, typography } from 'stylesheet';
import { flexGap } from 'services/cssHelpers';

import { Chip } from 'components/Chip';
import { Button } from 'components/Button';
import { Link } from 'components/Link';
import { LocalIconInformation, RemoteIconInformation } from 'components/Information';
import { TouristicContentDetailsType } from 'modules/touristicContent/interface';

import { Clock } from 'components/Icons/Clock';
import { CodeBrackets } from 'components/Icons/CodeBrackets';
import { TrendingUp } from 'components/Icons/TrendingUp';
import { ListAndMapContext } from 'modules/map/ListAndMapContext';

import { Attachment } from '../../../../../modules/interface';
import { dataUnits } from '../../../../../modules/results/adapter';
import { ResultCardCarousel } from './ResultCardCarousel';

interface BaseProps {
  id: string;
  hoverId: string | null;
  place: string | null;
  title: string;
  tags: string[];
  thumbnailUris: string[];
  redirectionUrl: string;
  attachments?: Attachment[];
  badgeIconUri?: string;
  badgeName?: string;
  className?: string;
}

interface TrekProps extends BaseProps {
  type: 'TREK';
  informations: {
    duration: string | null;
    distance: string;
    elevation: string;
    difficulty?: { label: string; pictogramUri: string } | null;
    reservationSystem: number | null;
  };
}

interface TouristicContentProps extends BaseProps {
  type: 'TOURISTIC_CONTENT';
  informations: TouristicContentDetailsType[];
}

interface OutdoorSiteProps extends BaseProps {
  type: 'OUTDOOR_SITE';
  informations: TouristicContentDetailsType[];
}

interface OutdoorCourseProps extends BaseProps {
  type: 'OUTDOOR_COURSE';
  informations: {
    duration: string | null;
    elevation?: string | null;
    maxElevation?: number;
    height: string | null;
    length: string | null;
  };
}

interface TouristicEventProps extends BaseProps {
  type: 'TOURISTIC_EVENT';
  informations: Record<any, any>;
}

const isTrek = (
  content:
    | TrekProps
    | TouristicContentProps
    | OutdoorSiteProps
    | OutdoorCourseProps
    | TouristicEventProps,
): content is TrekProps => content.type === 'TREK';

const isOutdoorCourse = (
  content:
    | TrekProps
    | TouristicContentProps
    | OutdoorSiteProps
    | OutdoorCourseProps
    | TouristicEventProps,
): content is OutdoorCourseProps => content.type === 'OUTDOOR_COURSE';

const isTouristicEvent = (
  content:
    | TrekProps
    | TouristicContentProps
    | OutdoorSiteProps
    | OutdoorCourseProps
    | TouristicEventProps,
): content is TouristicEventProps => content.type === 'TOURISTIC_EVENT';

export const ResultCard: React.FC<
  (
    | TrekProps
    | TouristicContentProps
    | OutdoorSiteProps
    | OutdoorCourseProps
    | TouristicEventProps
  ) & { asColumn?: boolean }
> = props => {
  const {
    id,
    hoverId,
    place,
    title,
    tags,
    thumbnailUris,
    attachments,
    badgeIconUri,
    badgeName,
    className,
    redirectionUrl,
    type,
    asColumn,
  } = props;
  const { setHoveredCardId } = useContext(ListAndMapContext);

  const intl = useIntl();

  return (
    <Container
      onMouseEnter={() => {
        setHoveredCardId(hoverId);
      }}
      onMouseLeave={() => {
        setHoveredCardId(null);
      }}
      className={className}
      id="result_card"
      asColumn={asColumn}
    >
      <Modal>
        {({ isFullscreen, toggleFullscreen }) => (
          <>
            {isFullscreen && attachments && attachments.length > 0 && (
              <DetailsCoverCarousel attachments={attachments} />
            )}
            {(!isFullscreen || !attachments) && (
              <ResultCardCarousel
                asColumn={asColumn}
                type={type}
                thumbnailUris={thumbnailUris}
                iconUri={badgeIconUri}
                iconName={badgeName as string}
                onClickImage={attachments ? toggleFullscreen : undefined}
              />
            )}
          </>
        )}
      </Modal>

      <Link href={redirectionUrl} testId={`Link-ResultCard-${id}`} className="w-full">
        <DetailsContainer>
          <DetailsLayout>
            {place !== null && <Place>{place}</Place>}

            <Title>{title}</Title>

            <TagContainer>
              <TagLayout>
                {tags
                  .filter(tag => tag !== null && Number(tag?.length) > 0)
                  .map(tag => (
                    <Chip key={tag}>{tag}</Chip>
                  ))}
              </TagLayout>
            </TagContainer>
            {isTouristicEvent(props) && (
              <InformationContainer>
                {props.informations.date && (
                  <LocalIconInformation icon={Calendar}>
                    {props.informations.date.beginDate === props.informations.date.endDate ? (
                      <FormattedMessage
                        id={'dates.singleDate'}
                        values={{ date: intl.formatDate(props.informations.date.beginDate) }}
                      />
                    ) : (
                      <FormattedMessage
                        id={'dates.multipleDates'}
                        values={{
                          beginDate: intl.formatDate(props.informations.date.beginDate),
                          endDate: intl.formatDate(props.informations.date.endDate),
                        }}
                      />
                    )}
                  </LocalIconInformation>
                )}
              </InformationContainer>
            )}
            {isOutdoorCourse(props) && (
              <InformationContainer>
                <InformationLayout>
                  {props.informations.duration && (
                    <LocalIconInformation icon={Clock}>
                      {props.informations.duration}
                    </LocalIconInformation>
                  )}
                  {props.informations.elevation && (
                    <LocalIconInformation icon={TrendingUp}>
                      {props.informations.elevation}
                    </LocalIconInformation>
                  )}
                  {props.informations.maxElevation && (
                    <LocalIconInformation icon={Altitude}>
                      {props.informations.maxElevation}
                      {dataUnits.distance}
                    </LocalIconInformation>
                  )}
                  {props.informations.length && (
                    <LocalIconInformation icon={CodeBrackets}>
                      {props.informations.length}
                    </LocalIconInformation>
                  )}
                  {props.informations.height && (
                    <LocalIconInformation icon={Height}>
                      {props.informations.height}
                    </LocalIconInformation>
                  )}
                </InformationLayout>
              </InformationContainer>
            )}
            {isTrek(props) && (
              <InformationContainer>
                <InformationLayout>
                  {props.informations.difficulty && (
                    <RemoteIconInformation iconUri={props.informations.difficulty.pictogramUri}>
                      {props.informations.difficulty.label}
                    </RemoteIconInformation>
                  )}
                  {props.informations.duration && (
                    <LocalIconInformation icon={Clock}>
                      {props.informations.duration}
                    </LocalIconInformation>
                  )}
                  {props.informations.distance && (
                    <LocalIconInformation icon={CodeBrackets}>
                      {props.informations.distance}
                    </LocalIconInformation>
                  )}
                  {props.informations.elevation && (
                    <LocalIconInformation icon={TrendingUp} className="desktop:flex hidden">
                      {props.informations.elevation}
                    </LocalIconInformation>
                  )}
                </InformationLayout>
              </InformationContainer>
            )}
            {!isTrek(props) && !isOutdoorCourse(props) && !isTouristicEvent(props) && (
              <InformationContainer>
                {props.informations.map(
                  ({ label, values }) =>
                    values.length > 0 && (
                      <div key={label} className="text-greyDarkColored">
                        <span className="font-bold">{`${label} : `}</span>
                        {values.join(', ')}
                      </div>
                    ),
                )}
              </InformationContainer>
            )}
          </DetailsLayout>
          {isTrek(props) &&
            props.informations.reservationSystem !== null &&
            false && ( // we disable this button because the booking behaviour is not implemented yet
              <BookingButtonContainer>
                <Button>
                  <FormattedMessage id="search.book" />
                </Button>
              </BookingButtonContainer>
            )}
        </DetailsContainer>
      </Link>
    </Container>
  );
};

const Container = styled.div<{ asColumn?: boolean }>`
  display: flex;
  flex: auto;
  flex-direction: column;
  cursor: pointer;
  border: 1px solid ${colorPalette.greySoft.DEFAULT};
  transition: all 500ms;
  &:hover {
    border-color: ${colorPalette.blackSemiTransparent};
  }
  border-radius: ${borderRadius.card};
  overflow: hidden;
  // Fix for overflow hidden with border radius in Safari, see https://gist.github.com/ayamflow/b602ab436ac9f05660d9c15190f4fd7b
  -webkit-mask-image: -webkit-radial-gradient(white, black);
  mask-image: radial-gradient(white, black);

  align-items: stretch;

  ${({ asColumn }) =>
    asColumn !== true &&
    desktopOnly(
      css`
        flex-direction: row;
      `,
    )}
`;

const DetailsContainer = styled.div`
  display: flex;
  width: 100%;

  padding: ${getSpacing(4)};

  ${desktopOnly(
    css`
      padding: ${getSpacing(6)};
    `,
  )}
`;

const DetailsLayout = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
`;

const BookingButtonContainer = styled.div`
  margin-left: ${getSpacing(4)};
  display: none;

  ${desktopOnly(
    css`
      display: block;
    `,
  )}
`;

const Place = styled.span`
  color: ${colorPalette.greyDarkColored};
  ${typography.small}
`;

const Title = styled.span`
  margin-top: ${getSpacing(1)};

  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;

  ${typography.h3}
  color: ${colorPalette.primary1};

  ${desktopOnly(
    css`
      text-overflow: clip;
      white-space: normal;
    `,
  )}
`;

const TagContainer = styled.div`
  display: flex;
  margin-top: ${getSpacing(2)};

  ${desktopOnly(
    css`
      margin-top: ${getSpacing(4)};
    `,
  )}
`;

const TagLayout = styled.div`
  ${flexGap(getSpacing(2))}

  ${desktopOnly(
    css`
      ${flexGap(getSpacing(4))}
    `,
  )}
`;

const InformationContainer = styled.div`
  margin-top: ${getSpacing(4)};
`;

const InformationLayout = styled.div`
  ${flexGap(getSpacing(2))}
  ${desktopOnly(
    css`
      ${flexGap(getSpacing(4))}
    `,
  )}
`;
