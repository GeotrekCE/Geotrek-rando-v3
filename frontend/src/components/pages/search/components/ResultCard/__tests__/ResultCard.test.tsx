import { render } from 'services/testing/reactTestingLibraryWrapper';

import { ResultCard } from '../ResultCard';

describe('Results Card', () => {
  const id = '2';
  const title = 'Balade au pays des menhirs';
  const urlToTest = `/details-${id}-Balade-au-pays-des-menhirs`;
  const resultCard = (
    <ResultCard
      id={id}
      hoverId="test"
      type="TREK"
      place="Saint-Etienne-du-Valdonnez"
      title={title}
      tags={['En famille', 'Ciel étoilé', 'Beau paysage']}
      attachments={[{ author: '', legend: '', url: '/test.jpg' }]}
      badgeIconUri=""
      informations={[
        {
          label: 'difficulty',
          value: 'Difficile',
          pictogramUri: '/difficulty.jpg',
        },
        { label: 'duration', value: '2h' },
        { label: 'distance', value: '5km' },
        { label: 'positiveElevation', value: '+360m' },
      ]}
      redirectionUrl={urlToTest}
    />
  );

  it('AAU, I can see a ResultCard', () => {
    const component = render(resultCard);
    expect(component).toMatchSnapshot();
  });

  it('AAU, I will be redirected to the right details page', () => {
    const component = render(resultCard);
    const link = component.queryByTestId(`Link-ResultCard-${id}`);
    expect(link).toHaveAttribute('href', urlToTest);
  });
});
