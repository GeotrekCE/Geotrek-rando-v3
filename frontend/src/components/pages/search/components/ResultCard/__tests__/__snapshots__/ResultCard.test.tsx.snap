// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results Card AAU, I can see a ResultCard 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ResultCard__Container-sc-1bem01n-0 jBsKrk"
        id="result_card"
      >
        <div
          style="background-color: white; position: relative;"
        >
          <div
            style="display: flex; align-items: center; justify-content: center; width: 100%; height: 100%;"
          >
            <div
              style="width: 100%; height: 100%;"
            >
              <div
                class="ResultCardCarousel__Wrapper-sc-13vjoai-0 botBuK h-full w-full flex-shrink-0 relative desktop:w-resultCardDesktop"
              >
                <div
                  class="slick-slider Carousel__StyledSlider-sc-1tck7ng-0 iyRBSY slick-initialized"
                >
                  
                  <div
                    class="slick-list"
                  >
                    <div
                      class="slick-track"
                      style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
                    >
                      <div
                        aria-hidden="false"
                        class="slick-slide slick-active slick-current"
                        data-index="0"
                        style="outline: none; width: 0px;"
                        tabindex="-1"
                      >
                        <div>
                          <div
                            class="relative h-full"
                            style="width: 100%; display: inline-block;"
                            tabindex="-1"
                          >
                            <img
                              alt=""
                              class="object-cover object-top w-full h-full"
                              src=""
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  
                  
                </div>
              </div>
            </div>
          </div>
        </div>
        <a
          class="Link__StyledLink-sc-oanbk8-0 htOOkH w-full min-h-0"
          data-testid="Link-ResultCard-2"
          href="/details-2-Balade-au-pays-des-menhirs"
        >
          <div
            class="ResultCard__DetailsContainer-sc-1bem01n-1 bUWpDM"
          >
            <div
              class="ResultCard__DetailsLayout-sc-1bem01n-2 gISiYd"
            >
              <span
                class="ResultCard__Place-sc-1bem01n-4 eJvZwY"
              >
                Saint-Etienne-du-Valdonnez
              </span>
              <span
                class="ResultCard__Title-sc-1bem01n-5 eBIDws"
              >
                Balade au pays des menhirs
              </span>
              <div
                class="ResultCard__TagContainer-sc-1bem01n-6 dhDoMi"
              >
                <div
                  class="ResultCard__TagLayout-sc-1bem01n-7 beobHM"
                >
                  <div
                    class="Chip__ChipContainer-sc-1wajafu-0 iTEhJ"
                  >
                    <span>
                      En famille
                    </span>
                  </div>
                  <div
                    class="Chip__ChipContainer-sc-1wajafu-0 iTEhJ"
                  >
                    <span>
                      Ciel étoilé
                    </span>
                  </div>
                  <div
                    class="Chip__ChipContainer-sc-1wajafu-0 iTEhJ"
                  >
                    <span>
                      Beau paysage
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="ResultCard__InformationContainer-sc-1bem01n-8 fxwsCG"
              >
                <div
                  class="ResultCard__InformationLayout-sc-1bem01n-9 cgxEzg"
                >
                  <div
                    class="flex items-center text-primary1  "
                  >
                    <img
                      class="object-cover object-center w-6 h-6"
                      src=""
                    />
                    <span
                      class="BaseInformation__InformationText-sc-1yqfii-0 bHCUEa"
                    />
                  </div>
                  <div
                    class="flex items-center text-primary1  "
                  >
                    <svg
                      fill="none"
                      height="24"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M12 22c5.523 0 10-4.477 10-10S17.523 2 12 2 2 6.477 2 12s4.477 10 10 10z"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                      />
                      <path
                        d="M12 6v6l4 2"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                      />
                    </svg>
                    <span
                      class="BaseInformation__InformationText-sc-1yqfii-0 bHCUEa"
                    >
                      2h
                    </span>
                  </div>
                  <div
                    class="flex items-center text-primary1  "
                  >
                    <svg
                      fill="none"
                      height="24"
                      viewBox="0 0 24 25"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M16 18l6-6-6-6M8 6l-6 6 6 6"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                      />
                    </svg>
                    <span
                      class="BaseInformation__InformationText-sc-1yqfii-0 bHCUEa"
                    >
                      5km
                    </span>
                  </div>
                  <div
                    class="flex items-center text-primary1 desktop:flex hidden "
                  >
                    <svg
                      fill="none"
                      height="24"
                      viewBox="0 0 24 25"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M23 6l-9.5 9.5-5-5L1 18"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                      />
                      <path
                        d="M17 6h6v6"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                      />
                    </svg>
                    <span
                      class="BaseInformation__InformationText-sc-1yqfii-0 bHCUEa"
                    >
                      +360m
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ResultCard__Container-sc-1bem01n-0 jBsKrk"
      id="result_card"
    >
      <div
        style="background-color: white; position: relative;"
      >
        <div
          style="display: flex; align-items: center; justify-content: center; width: 100%; height: 100%;"
        >
          <div
            style="width: 100%; height: 100%;"
          >
            <div
              class="ResultCardCarousel__Wrapper-sc-13vjoai-0 botBuK h-full w-full flex-shrink-0 relative desktop:w-resultCardDesktop"
            >
              <div
                class="slick-slider Carousel__StyledSlider-sc-1tck7ng-0 iyRBSY slick-initialized"
              >
                
                <div
                  class="slick-list"
                >
                  <div
                    class="slick-track"
                    style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
                  >
                    <div
                      aria-hidden="false"
                      class="slick-slide slick-active slick-current"
                      data-index="0"
                      style="outline: none; width: 0px;"
                      tabindex="-1"
                    >
                      <div>
                        <div
                          class="relative h-full"
                          style="width: 100%; display: inline-block;"
                          tabindex="-1"
                        >
                          <img
                            alt=""
                            class="object-cover object-top w-full h-full"
                            src=""
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                
                
              </div>
            </div>
          </div>
        </div>
      </div>
      <a
        class="Link__StyledLink-sc-oanbk8-0 htOOkH w-full min-h-0"
        data-testid="Link-ResultCard-2"
        href="/details-2-Balade-au-pays-des-menhirs"
      >
        <div
          class="ResultCard__DetailsContainer-sc-1bem01n-1 bUWpDM"
        >
          <div
            class="ResultCard__DetailsLayout-sc-1bem01n-2 gISiYd"
          >
            <span
              class="ResultCard__Place-sc-1bem01n-4 eJvZwY"
            >
              Saint-Etienne-du-Valdonnez
            </span>
            <span
              class="ResultCard__Title-sc-1bem01n-5 eBIDws"
            >
              Balade au pays des menhirs
            </span>
            <div
              class="ResultCard__TagContainer-sc-1bem01n-6 dhDoMi"
            >
              <div
                class="ResultCard__TagLayout-sc-1bem01n-7 beobHM"
              >
                <div
                  class="Chip__ChipContainer-sc-1wajafu-0 iTEhJ"
                >
                  <span>
                    En famille
                  </span>
                </div>
                <div
                  class="Chip__ChipContainer-sc-1wajafu-0 iTEhJ"
                >
                  <span>
                    Ciel étoilé
                  </span>
                </div>
                <div
                  class="Chip__ChipContainer-sc-1wajafu-0 iTEhJ"
                >
                  <span>
                    Beau paysage
                  </span>
                </div>
              </div>
            </div>
            <div
              class="ResultCard__InformationContainer-sc-1bem01n-8 fxwsCG"
            >
              <div
                class="ResultCard__InformationLayout-sc-1bem01n-9 cgxEzg"
              >
                <div
                  class="flex items-center text-primary1  "
                >
                  <img
                    class="object-cover object-center w-6 h-6"
                    src=""
                  />
                  <span
                    class="BaseInformation__InformationText-sc-1yqfii-0 bHCUEa"
                  />
                </div>
                <div
                  class="flex items-center text-primary1  "
                >
                  <svg
                    fill="none"
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M12 22c5.523 0 10-4.477 10-10S17.523 2 12 2 2 6.477 2 12s4.477 10 10 10z"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                    />
                    <path
                      d="M12 6v6l4 2"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                    />
                  </svg>
                  <span
                    class="BaseInformation__InformationText-sc-1yqfii-0 bHCUEa"
                  >
                    2h
                  </span>
                </div>
                <div
                  class="flex items-center text-primary1  "
                >
                  <svg
                    fill="none"
                    height="24"
                    viewBox="0 0 24 25"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M16 18l6-6-6-6M8 6l-6 6 6 6"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                    />
                  </svg>
                  <span
                    class="BaseInformation__InformationText-sc-1yqfii-0 bHCUEa"
                  >
                    5km
                  </span>
                </div>
                <div
                  class="flex items-center text-primary1 desktop:flex hidden "
                >
                  <svg
                    fill="none"
                    height="24"
                    viewBox="0 0 24 25"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M23 6l-9.5 9.5-5-5L1 18"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                    />
                    <path
                      d="M17 6h6v6"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                    />
                  </svg>
                  <span
                    class="BaseInformation__InformationText-sc-1yqfii-0 bHCUEa"
                  >
                    +360m
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
